setup.py

```diff
@@ -137,23 +136,11 @@ def rm(file):
         pass
 
 
-class Clean(clean):
-    def run(self):
-        super().run()
-        for source in cython_sources:
-            genc = source.replace('.pyx', '.c')
-            rm(genc)
-            compiled_glob = source.replace('.pyx', '.cpython*')
-            for compiled in sorted(glob(compiled_glob)):
-                rm(compiled)
```

Comment:
@RonnyPfannschmidt I would like to keep that, but did not find a way yet to do this without importing from distutils.

any idea?

Reply:
Unfortunately not, perhaps @jaraco has a better idea

Reply:
Setuptools is generally discouraging the use of anything `setup.py <cmd>`, which would include `setup.py clean` in favor of isolated builds using PEP 517 builders (e.g. build).

That said, Setuptools is adopting distutils and has vendored this clean command, so will supply it for now. Until its integrated naturally into Setuptools, however, it will be difficult to extend. I'm okay with the proposed approach to provide an extra command.